### PR TITLE
Fix nvJPEGDecoder unit tests

### PR DESCRIPTION
**Pull request by jantonguirao**
Signed-off-by: Joaquin Anton <janton@nvidia.com>

### CODE REVIEW

File: dali/test/dali_test_decoder.h [hunk 1]
```diff
@@ -109,6 +109,10 @@ class GenericDecoderTest : public DALISingleOpTest<ImgType> {
     this->DecodeImage(imgData, imgSize, c, this->ImageType(), &out);
     this->CheckBuffers(h * w * c, out.mutable_data<uint8>(), img, false);
   }
+
+  uint32_t GetImageLoadingFlags() const override {
+    return t_loadJPEGs + t_loadPNGs + t_loadTiffs;
```
Comment (Kh4L) [inline dali/test/dali_test_decoder.h]: bitwise op?

File: dali/test/dali_test_single_op.h [hunk 2]
```diff
@@ -15,15 +15,13 @@
 #ifndef DALI_TEST_DALI_TEST_SINGLE_OP_H_
 #define DALI_TEST_DALI_TEST_SINGLE_OP_H_
 
-#include "dali/test/dali_test.h"
-
+#include <dirent.h>
```
Comment (Kh4L) [inline dali/test/dali_test_single_op.h]: Why this include?

**Comment by jantonguirao** [comment]
Build #653726

File: dali/pipeline/operators/decoder/host_decoder_test.h [hunk 3]
```diff
@@ -39,7 +39,7 @@ class HostDecodeTestBase : public GenericDecoderTest<ImgType> {
   }
 
   inline uint32_t GetImageLoadingFlags() const override {
-    return image_type_;
+    return t_loadJPEGs | t_loadPNGs | t_loadTiffs;
```
Comment (Kh4L) [inline dali/pipeline/operators/decoder/host_decoder_test.h]: Can't we just get rid of this function?

Comment (jantonguirao) [reply dali/pipeline/operators/decoder/host_decoder_test.h]: we can get rid of so many things :), but I don't want to waste more time in grooming those tests now

**Pull request merged**